community/kernel/src/main/java/org/neo4j/kernel/impl/api/StateHandlingStatementOperations.java

```diff
@@ -742,112 +732,105 @@ private void indexUpdateProperty( KernelStatement state, long nodeId, int labelI
     public PrimitiveIntIterator nodeGetPropertyKeys( KernelStatement state, long nodeId )
             throws EntityNotFoundException
     {
-        if ( state.hasTxStateWithChanges() )
+        try ( NodeCursor nodeCursor = acquireSingleNodeCursor( state, nodeId ) )
         {
-            return new PropertyKeyIdIterator( nodeGetAllProperties( state, nodeId ) );
-        }
+            if ( !nodeCursor.next() )
+            {
+                throw new EntityNotFoundException( EntityType.NODE, nodeId );
```

Comment:
Almost the same story here... [`nodeGetLabels`] (https://github.com/neo4j/neo4j/pull/4848/files#diff-94e898a8535f13220449b217db75fd90R260) will return empty iterator instead of exception.